### ISSUE TITLE
Unknown CHPL_TARGET_COMPILER/CHPL_TARGET_CC with 1.25.1

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#18973]
While setting up a Chapel installation of 1.25.1 on Ookami, I ran into the following points of confusion:

* I built with CHPL_LLVM=none and CHPL_TARGET_PLATFORM=hpe-apollo (though I'd guess this second setting is immaterial?)
* After building, I set up a `chplconfig` file as follows, to match what I'd set when building:
  ```
  CHPL_TARGET_PLATFORM=hpe-apollo
  CHPL_TARGET_CPU=none
  ```
* A user who happened to have `CHPL_TARGET_COMPILER=gnu` set in their shell tried using my build and it failed, despite having been built with gcc.  Unsetting that permitted it to work again.  I think the following observations are the "reason."
* Adding `CHPL_LLVM=none` to the `chplconfig` file caused the output of `printchplenv` to give no value for `CHPL_TARGET_COMPILER`, which is unusual behavior to me (and seems more like an error condition than a "print nothing" one):
  ```
  ...
  CHPL_TARGET_PLATFORM: hpe-apollo +
  CHPL_TARGET_COMPILER: 
  CHPL_TARGET_ARCH: aarch64
  ...
  ```
* Running `chpl --print-chpl-settings` indicated a similar confusion:
  ```
  CHPL_TARGET_COMPILER: 
    CHPL_TARGET_CC: unknown-c-compiler
    CHPL_TARGET_CXX: unknown-c++-compiler
 ```

My guess is that the `build` directories specialized on the fact that the C compiler was "unknown" rather than inferring the target compiler to be gcc and using that as the string (which would both fill in the blanks in these final output steps, and would also mean that the user who happened to have `gnu` set wouldn't have had problems).

I have not had the chance to try and reproduce this with `main` yet to see whether the behavior has changed or improved since the last release.

**Comment by bradcray** [comment]
@mppf: Checking to see whether this behavior seems surprising/wrong to you as well?  (Specifically the lack of a correctly inferred CHPL_TARGET_COMPILER)

**Comment by mppf** [comment]
Yes that seems surprising/wrong - we should either have an error emitted or a non-empty CHPL_TARGET_COMPILER inferred
